upgrade.php

```diff
@@ -130,36 +130,90 @@ function getTableIndexes($table)
     */
 
     $server_utf8_variant = 'utf8';
-    if ($link->server_version >= 50503) {
-        $server_utf8_variant = 'utf8mb4';
-    }
 
-    echo pad(' - Convert database to UTF-8');
+    echo pad(' - Convert database to utf8');
     $sql = 'ALTER DATABASE `' . DB_NAME .
         '` CHARACTER SET = ' . $mysql_utf8_variant[$server_utf8_variant]['charset'] .
         ' COLLATE = ' . $mysql_utf8_variant[$server_utf8_variant]['collation'];
     executeQuery($sql);
 
+    // Truncate needed for VARCHAR field used as PRIMARY or FOREIGN KEY when using UTF-8mb4
+
+    // Table users
+    echo pad(' - Fix schema for username field in `users` table');
+    $sql = "ALTER TABLE `users` CHANGE `username` `username` VARCHAR( 191 ) NOT NULL DEFAULT ''";
+    executeQuery($sql);
+
+    // Table spamscores
+    echo pad(' - Fix schema for user field in `spamscores` table');
+    $sql = "ALTER TABLE `spamscores` CHANGE `user` `user` VARCHAR( 191 ) NOT NULL DEFAULT ''";
+    executeQuery($sql);
+
+    // Table user_filters
+    echo pad(' - Fix schema for username field in `user_filters` table');
+    $sql = "ALTER TABLE `user_filters` CHANGE `username` `username` VARCHAR( 191 ) NOT NULL DEFAULT ''";
+    executeQuery($sql);
+    
+    // Revert back some tables to the right values due to previous errors in upgrade.php
```

Comment:
If username has to be 191 chars at max the "right" values is always 191, not more: every reference of username in tables has to be 191 chars long